### ISSUE TITLE
Make tools for building sys161 BF images

### ISSUE THREAD

**Issue by penlu** [Harvard-PRINCESS/Guppy#102]
child of #99 

We're not sure what these tools will be yet. Shell scripts to pack compiler binary outputs into ELF or some other boot image format presumably.

**Comment by ghost** [comment]
After discussion this afternoon:
* It seems like the path of least resistance is to link the bfish kernel image and the modules to be loaded with it into a single elf file to be loaded by sys161 in the usual way. There's already some code for this in the barrelfish tree, for some old arm thing, and adjusting it for this environment should only take minimal changes.
* I'm open to providing a way to have sys161 load both a kernel image and a collection of modules, since it's something anyone doing microkernel stuff on sys161 will want. But it's not clear how to do this in a way that's both simple and effective, either on the user-facing command-line side or the kernel-facing startup-time side. I'll think about it. If anyone has ideas, bring them along.

**Comment by alexpatel** [comment]
> There's already some code for this in the barrelfish tree, for some old arm thing, and adjusting it for this environment should only take minimal changes.

See [tools/arm_boot](https://github.com/Harvard-PRINCESS/Guppy/tree/port-mips-sys161/tools/arm_boot)

**Comment by marikgoldstein** [comment]
checklist as of 8/9/2017:
- adapt static bootloader for MIPS (which prepares an ELF boot image) (equivalent of tools/arm_boot)
- add compiler binary names to hake/MIPS.hs, and also LD script stuff
- modify kernel/Hakefile to produce MIPS boot driver (like boot ve)
- modify platforms/Hakefile to use static bootloader to make image (make a mipsImage equivalent of armv7Image)
- stub out start.S, exception.S
- boot rule: run sys161! Add this to the series of boot commands in platforms/Hakefile.

Basically, mips_boot.c and a corresponding Hakefile in tools/mips_boot should create a static bootloader image (say, called mips_boot_loader). A Hakefile in kernel/ should make a bootdriver image. The bootloader and bootdriver images should be passed to mipsImage() in platforms/Hakefile.

**Comment by penlu** [comment]
another thing on the map: adapting the BF linker system to do what os161 link is doing, but on the barrelfish-ized files
